operation/manager_test.go

```diff
@@ -155,24 +172,6 @@ func TestErrorContainsStopAndCancel(t *testing.T) {
 	test.That(t, errRet.Error(), test.ShouldEqual, "context canceled; Stop failed")
 }
 
-func TestStopNotCalledOnOldContext(t *testing.T) {
-	som := SingleOperationManager{}
-	ctx, done := som.New(context.Background())
-	defer done()
-	mock := &mock{stopCount: 0}
-	var wg sync.WaitGroup
-
-	wg.Add(1)
-	go func() {
-		som.WaitTillNotPowered(ctx, time.Second, mock, mock.stop)
-		wg.Done()
-	}()
-	som.New(context.Background())
```

Comment:
This test relied on canceling not blocking (i.e: the "canceler" doing this line 170 `som.New` call would change the `sm.currentOp` to the new operation before the cancelation wakes up the old operation).

In the new world this `New` would wait for cancelation to happen first. Thus I [expect `oldOp` to always be true](https://github.com/viamrobotics/rdk/blob/572ed28e85cac31f2266a23cec1b66163e60e210/operation/manager.go#L102-L104) for this case resulting in a call to `stop`. In the next patch, I've added a comment in `WaitTillNotPowered` so a future author can more carefully review if the conditional is necessary.